ktor-http/ktor-http-cio/src/io/ktor/http/cio/internals/WeakTimeoutQueue.kt

```diff
@@ -142,6 +142,7 @@ class WeakTimeoutQueue(
         }
     }
 
+    @UseExperimental(InternalCoroutinesApi::class)
     private class WeakTimeoutCoroutine<in T>(context: CoroutineContext, val delegate: Continuation<T>) :
```

Comment:
`WeakTimeoutCoroutine` has probably the same bugs as `TimeoutCoroutine` had.
I've tried to comment some parts of cancellation machinery here and there, but none of `cio` tests have failed, so I decided not to touch this code.

Reply:
Yes, this need to be updated to be in-sync with `TimeoutCoroutine`